### ISSUE TITLE
Error while saving sitemap

### ISSUE THREAD

**Issue by wateengoedidee** [ethercreative/seo#20]
In the SEO -> Sitemap plugin settings, while saving the sitemap an error occurred:

Invalid argument supplied for foreach()

/var/www/craft/plugins/seo/services/Seo_SitemapService.php(40)

```
$oldSitemap = $this->getSitemap();
$newSitemap = $data;

 // Delete removed rows
 $newById = [];
 $oldById = [];

 $newRecordsRaw = [];

 foreach ($newSitemap as $group => $rows) {
     foreach ($rows as $new) {  // This is the affected row (40)
         $new['group'] = $group;
         if ($new['id'] != "-1") {
             $newById[$new['id']] = $new;
         } else {
             $newRecordsRaw[] = $new;
         }
     }
 }

 $idsToDelete = [];
 foreach ($oldSitemap as $group => $rows) {
     foreach ($rows as $old)
```

**Comment by roelvanhintum** [comment]
Just got the same error after updating SEO from 1.1.3 to 1.2.3

**Comment by ClemensSchneider** [comment]
Can confirm this issue. Although it's just a PHP warning. Disabling `devMode` allows me to save the changes.

**Comment by PetterRuud** [comment]
Simple fix for now
`		foreach ($newSitemap as $group => $rows)
		{
  		
			foreach ( (array)$rows as $new)
			{
				if (!empty($new['id']) && $new['id'] != null) {
					$new['group'] = $group;
	
					if ($new['id'] != "-1") $newById[$new['id']] = $new;
					else $newRecordsRaw[] = $new;
				}
			}
		}`

**Comment by Tam** [comment]
Should be fixed in v1.3.0? I haven't been able to re-create it, so let me know if it's not fixed!

**Comment by Tam** [comment]
Okay, so I can re-create it! This should actually be fixed in v1.3.1

**Comment by wateengoedidee** [comment]
Ah Thanks ;)




Met vriendelijke groet,

Tom Fischer

Bezoekadres Het Eeftink 11-21
7541 WH Enschede
T +316 24 89 41 19
E info@wateengoedidee.nl
W www.wateengoedidee.nl

Op 9 november 2016 bij 13:25:23, Tam (notifications@github.com) schreef:

Okay, so I can re-create it! This should actually be fixed in v1.3.1

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/ethercreative/seo/issues/20#issuecomment-259403138>,
or mute
the thread
